initializers/database.go

```diff
@@ -60,24 +64,77 @@ func connectSqlite() (*gorm.DB, error) {
 }
 
 func connectPostgres() (*gorm.DB, error) {
-	dsn := fmt.Sprintf(
-		"host=%s user=%s password=%s dbname=%s port=%d %s",
-		config.DB_POSTGRES_HOST,
-		config.DB_POSTGRES_USERNAME,
-		config.DB_POSTGRES_PASSWORD,
-		config.DB_POSTGRES_DB_NAME,
-		config.DB_POSTGRES_PORT,
-		config.DB_POSTGRES_DSN_EXTRA,
-	)
+	var dsn string
+	if config.DB_POSTGRES_DSN != "" {
+		dsn = config.DB_POSTGRES_DSN
+	} else {
+		dsn = fmt.Sprintf(
+			"host=%s user=%s password=%s dbname=%s port=%d %s",
```

Comment:
yeah it's more streamlined to just pass the uri

Reply:
Done.